src/main/groovy/com/google/protobuf/gradle/ProtobufPlugin.groovy

```diff
@@ -114,8 +114,8 @@ class ProtobufPlugin implements Plugin<Project> {
     }
 
     @TypeChecked(TypeCheckingMode.SKIP)
-    private static void linkGenerateProtoTasksToTask(Task task, GenerateProtoTask genProtoTask) {
-      task.source genProtoTask.getOutputSourceDirectorySet().include("**/*.java", "**/*.kt")
+    private static void linkGenerateProtoTasksToTask(Task task, Provider<GenerateProtoTask> genProtoTask) {
```

Comment:
`Provider<GenerateProtoTask>` -> `TaskProvider<GenerateProtoTask>`
Gradle has a special provider class for tasks.

Reply:
The only additional method the TaskProvider has is the name, which we really shouldn't be needing. Certainly here where it is being passed in as a parameter, the lowest-level interface is preferable.